pkg/streamingpromql/planning.go

```diff
@@ -441,7 +441,7 @@ func findFunction(name string) (functions.Function, bool) {
 // Materialize converts a query plan into an executable query.
 func (e *Engine) Materialize(ctx context.Context, plan *planning.QueryPlan, queryable storage.Queryable, opts promql.QueryOpts) (promql.Query, error) {
 	if opts == nil {
-		opts = promql.NewPrometheusQueryOpts(false, 0)
+		opts = promql.NewPrometheusQueryOpts(false, 0, model.UTF8Validation)
```

Comment:
> @juliusmh Should q.nameValidationScheme be used here?

What is `q` ? At this point we don't have a query yet. We could give the engine `e` a configurable default. FYI, prometheus uses global `model.NameValidationScheme` here.